docs/index.rst

```diff
@@ -29,7 +29,7 @@ Contributors
 ------------
 
 A list of contributors for the analysis pipeline is available `here <https://github.com/spine-generic/spine-generic/graphs/contributors>`__.
-If you would like to contribute to this project, please see `contribution guidelines <https://github.com/spine-generic/spine-generic/blob/master/CONTRIBUTING.md>`_.
```

Comment:
I'm curious about the motivation for each of these style changes. I presumed that they would be necessary to placate Sphinx's fail_on_warning setting, yet it seems to be absent from the new  `.readthedocs.yaml` file?

Is that something we want to add, as was done [for SCT](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/45412456dfb7a54542e1eb8aa0eb6ca8ffbb44db/.readthedocs.yaml#L25-L28)? Or are there still too many warnings to want to turn that on?

Reply:
They were just warnings from my local Sphinx build, or inspired from what I learned by fixing warnings, not from any automated checks. A short explanation:
- *For changing links from `_` (single underscore) to `__` (double underscore):* Apparently using a single underscore at the end of a hyperlink, like
  <pre><code>`my link text &lt;https://example.com/my/link/url.html>`_</code></pre>
  creates a document-wide link target, keyed by `my link text` (see [here](https://docutils.sourceforge.io/docs/user/rst/quickref.html#hyperlink-targets), maybe? The documentation is not great). And Sphinx doesn't want multiple targets with the same text, which generated a warning. To avoid this, the advice is to use double underscore, which I tried to apply uniformly to all links.
- *For the whitespace in a table row:* There was a mismatch with the surrounding lines, and Sphinx actually complained about it, so I fixed it. Possibly the result of a search-and-replace that didn't take into account word lengths.
- *For removing some of the header link targets:* Most of them were unused? Maybe I could have left them in. But I think the rule should probably be either "Only used targets get defined" or "All headers get targets", and I didn't feel like adding all those targets. This was triggered because one of the used targets wasn't defined, causing a warning (and a dead link).

Reply:
> Ah, I just realised that SG doesn't have RTD building set up for PRs, so the utility of fail_on_warning (CI checks) is lost.

Yeah, actually setting that up seemed out of scope.